src/commands/merge.rs

```diff
@@ -16,12 +22,115 @@ pub struct MergeCommandArgs {
     /// Parquet file to write
     #[arg(short, long)]
     output: PathBuf,
+
+    /// Path to a json config file specifying WriterProperties::builder() properties.
+    #[arg(short, long)]
+    config: Option<PathBuf>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct MergeConfig {
+    pub set_dictionary_enabled: Option<bool>,
+    /// The encodings for this are the just text values of the enum parquet::basic::Encoding
+    pub column_encodings: Option<HashMap<String, String>>,
+    pub column_dictionary_enabled: Option<HashMap<String, bool>>,
+    pub compression: Option<String>,
+    pub compression_level: Option<u32>,
+}
+
+fn build_encoding_mappings() -> HashMap<&'static str, Encoding> {
+    HashMap::from([
+        ("PLAIN", Encoding::PLAIN),
+        ("PLAIN_DICTIONARY", Encoding::PLAIN_DICTIONARY),
+        ("RLE", Encoding::RLE),
+        ("BIT_PACKED", Encoding::BIT_PACKED),
+        ("DELTA_BINARY_PACKED", Encoding::DELTA_BINARY_PACKED),
+        ("DELTA_LENGTH_BYTE_ARRAY", Encoding::DELTA_LENGTH_BYTE_ARRAY),
+        ("DELTA_BYTE_ARRAY", Encoding::DELTA_BYTE_ARRAY),
+        ("RLE_DICTIONARY", Encoding::RLE_DICTIONARY),
+        ("BYTE_STREAM_SPLIT", Encoding::BYTE_STREAM_SPLIT),
+    ])
+}
+
+fn build_props_from_json_config(
+    config_path: PathBuf,
+) -> Result<WriterProperties, PQRSError> {
+    let data = fs::read_to_string(config_path)?;
+    let merge_config: MergeConfig = serde_json::from_str(&data)?;
+    let mut props =
+        WriterProperties::builder().set_writer_version(WriterVersion::PARQUET_2_0);
+
+    if let Some(de) = merge_config.set_dictionary_enabled {
+        props = props.set_dictionary_enabled(de);
+    }
+
+    if let Some(column_encodings) = merge_config.column_encodings {
+        let encoding_mappings = build_encoding_mappings();
+        for (column_name, encoding_str) in column_encodings {
+            if !encoding_mappings.contains_key(encoding_str.as_str()) {
+                return Err(PQRSError::IllegalEncodingType());
+            }
+
+            let encoding = *encoding_mappings
+                .get(encoding_str.clone().as_str())
+                .unwrap();
+            props = props.set_column_encoding(ColumnPath::from(column_name), encoding)
+        }
+    }
+
+    if let Some(column_de) = merge_config.column_dictionary_enabled {
+        for (column_name, de) in column_de {
+            println!("{column_name}");
+            props =
+                props.set_column_dictionary_enabled(ColumnPath::from(column_name), de);
+        }
+    }
+
+    if let Some(compression_algo) = merge_config.compression {
+        if compression_algo.to_lowercase() == "brotli" {
```

Comment:
same for `compression_algo`, I think we can use pattern matching here?

Reply:
Good callout!